gemfire-core/src/main/java/com/gemstone/gemfire/internal/ByteBufferDataOutput.java

```diff
@@ -67,10 +67,9 @@ public ByteBufferDataOutput(Version version) {
    * flag to UMM.
    */
   public ByteBufferDataOutput(int initialSize,
-      BufferAllocator allocator,
       Version version,
       String bufferOwner) {
-    this.allocator = allocator;
+    this.allocator = GemFireCacheImpl.getCurrentBufferAllocator();
```

Comment:
Should not be done here rather by caller. As mentioned there, I see a problem with using heap allocator for result sending.